apps/sv/frontend/src/components/votes/actions/CreateUnallocatedUnclaimedActivityRecord.tsx

```diff
@@ -0,0 +1,163 @@
+import { ActionFromForm, actionFromFormIsError } from '../VoteRequest';
+
+dayjs.extend(utc);
+
+function asCreateUnallocatedUnclaimedActivityRecord(action?: ActionFromForm):
+  | {
+      beneficiary: string;
+      amount: string;
+      reason: string;
+      expiresAt: string;
+    }
+  | undefined {
+  return action &&
+    !actionFromFormIsError(action) &&
+    action.tag === 'ARC_DsoRules' &&
+    action.value.dsoAction.tag === 'SRARC_CreateUnallocatedUnclaimedActivityRecord'
+    ? action.value.dsoAction.value
+    : undefined;
+}
+
+const CreateUnallocatedUnclaimedActivityRecord: React.FC<{
+  chooseAction: (action: ActionRequiringConfirmation) => void;
+  action?: ActionFromForm;
+  effectivity: Dayjs;
+}> = ({ chooseAction, action, effectivity }) => {
+  const existing = asCreateUnallocatedUnclaimedActivityRecord(action);
+
+  const [beneficiary, setBeneficiary] = useState(existing?.beneficiary ?? '');
+  const [amount, setAmount] = useState(existing?.amount ?? '');
+  const [reason, setReason] = useState(existing?.reason ?? '');
+  const [expiresAt, setExpiresAt] = useState<Dayjs>(dayjs());
+  const [amountError, setAmountError] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (existing?.expiresAt) {
+      setExpiresAt(dayjs(existing.expiresAt));
+    } else {
+      setExpiresAt(effectivity.add(1, 'day'));
+    }
+  }, [existing?.expiresAt, effectivity]);
+
+  const updateAction = useCallback(
+    (beneficiary: string, amount: string, reason: string, expiresAt: Dayjs) => {
+      chooseAction({
+        tag: 'ARC_DsoRules',
+        value: {
+          dsoAction: {
+            tag: 'SRARC_CreateUnallocatedUnclaimedActivityRecord',
+            value: {
+              beneficiary,
+              amount,
+              reason,
+              expiresAt: expiresAt.toISOString(),
+            },
+          },
+        },
+      });
+    },
+    [chooseAction]
+  );
+
+  useEffect(() => {
+    if (beneficiary && amount && reason && expiresAt && !amountError) {
+      updateAction(beneficiary, amount, reason, expiresAt);
+    }
+  }, [beneficiary, amount, reason, expiresAt, amountError, updateAction]);
+
+  return (
+    <Stack direction="column" mb={4} spacing={1}>
+      <Typography variant="h6">Beneficiary</Typography>
+      <FormControl fullWidth>
+        <TextField
+          id="create-beneficiary"
+          inputProps={{ 'data-testid': 'create-beneficiary' }}
+          value={beneficiary}
+          onChange={e => setBeneficiary(e.target.value)}
+        />
+      </FormControl>
+
+      <Typography variant="h6">Amount</Typography>
+      <FormControl fullWidth>
+        <TextField
+          id="create-amount"
+          type="number"
+          value={amount}
+          inputProps={{ 'data-testid': 'create-amount' }}
+          onChange={e => {
+            const newValue = e.target.value;
+            setAmount(newValue);
+
+            // TODO: Check that amount <=  amount of unclaimed rewards available.
```

Comment:
```suggestion
            // TODO(#1520): Check that amount <=  amount of unclaimed rewards available.
```
todos must always be associated with an issue